.github/workflows/.terraform.yaml

```diff
@@ -0,0 +1,216 @@
+    inputs:
+      workload_identity_provider:
+        type: string
+        required: true
+        description: "Workload identity provider for GCP Workload Federation."
+      service_account:
+        type: string
+        required: true
+        description: "GCP Service Account to impersonate."
+      checkout_sha:
+        required: false
+        type: string
+        description: 'Commit SHA to run at.'
+      working_directory:
+        required: true
+        type: string
+        description: 'Location for the terraform.'
+      project_id:
+        required: false
+        type: string
+        description: 'GCP project ID.'
+      octo-sts-identity:
+        required: false
+        default: ''
+        type: string
+        description: 'The Octo STS identity name'
+
+jobs:
+  terraform:
+    name: Terraform
+    runs-on: 'ubuntu-latest'
+    permissions:
+      contents: read  # clone the repository contents
+      id-token: write # federates with GCP
+
+    env:
+      SHA: ${{ inputs.checkout_sha || github.sha }}
+      PROJECT_ID: ${{ inputs.project_id }}
+      TF_PLAN_BIN: 'plan.tmp'
+      TF_PLAN_OUT: 'plan.out'
+
+    defaults:
+      run:
+        working-directory: "${{ inputs.working_directory }}"
+
+    steps:
+      - uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        with:
+          egress-policy: audit
+
+      - name: Checkout
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          ref: ${{ env.SHA }}
+
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version-file: './go.mod'
+          check-latest: true
+
+      - name: Authenticate to Google Cloud
+        uses: 'google-github-actions/auth@55bd3a7c6e2ae7cf1877fd1ccb9d54c0503c457c' # v2.1.2
+        id: auth
+        with:
+          token_format: 'access_token'
+          project_id: '${{ inputs.project_id }}'
+          workload_identity_provider: '${{ inputs.workload_identity_provider }}'
+          service_account: '${{ inputs.service_account }}'
+
+      - name: 'Set up Cloud SDK'
+        uses: google-github-actions/setup-gcloud@98ddc00a17442e89a24bbf282954a3b65ce6d200 # v2.1.0
+
+      - uses: 'docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20' # v2
+        with:
+          username: 'oauth2accesstoken'
+          password: '${{ steps.auth.outputs.access_token }}'
+          registry: 'gcr.io'
+
+      - name: Setup terraform
+        uses: 'hashicorp/setup-terraform@a1502cd9e758c50496cc9ac5308c4843bcd56d36' # v3.0.0
+        with:
+          terraform_version: 1.6
+
+      - name: Terraform fmt
+        id: fmt
+        run: terraform fmt -check -recursive
+
+      - name: Terraform init
+        id: init
+        run: terraform init -reconfigure
+
+      - name: Terraform Validate
+        id: validate
+        run: terraform validate -no-color
+
+      - name: Terraform Plan
```

Comment:
Do you know of anything that we can run over this to check for injection vulnerabilities?

Skimming, the only one that stood out is `inputs.working_directory`, and that's a reach, but would be good to sanity check things